File: 102-BinaryTreeLevelOrderTraversal/102-BinaryTreeLevelOrderTraversal.java
// Last updated: 8/13/2025, 8:22:53 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> q = new LinkedList();
        List<List<Integer>> list = new ArrayList();
        q.add(root);
        if(root == null) return list;
        while(q.size()>0){
            List<Integer> sublist = new ArrayList();
            int n = q.size();
            while(n>0){
                TreeNode temp = q.remove();
                sublist.add(temp.val);
                if(temp.left!=null){
                    q.add(temp.left);
                }
                if(temp.right!=null){
                    q.add(temp.right);
                }
                n--;
            }
            list.add(sublist);
        }
        return list;
    }
}

File: 102-BinaryTreeLevelOrderTraversal/102-BinaryTreeLevelOrderTraversal.py
# Last updated: 8/13/2025, 8:22:47 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        my_queue = []
        result = []
        if root is None : 
            return []
        my_queue.append(root)

        while len(my_queue) > 0 :
            n = len(my_queue)
            temp_list = []

            while n > 0 : 
                temp_node = my_queue.pop(0)
                temp_list.append(temp_node.val)

                if temp_node.left : 
                    my_queue.append(temp_node.left)
                if temp_node.right :
                    my_queue.append(temp_node.right)
                
                n-=1
            result.append(temp_list)
        return result